### PR TITLE
fix(n-split): modified `default-size` panel not updated issue

### PR DESCRIPTION
**Pull request by jahnli**
close https://github.com/tusen-ai/naive-ui/issues/5526

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -26,6 +26,7 @@
 - Fix `n-split` has no color if it's not used in a card.
 - Fix `n-card` `footer-class` prop not working.
 - Fix `n-tabs` may miss over-scroll shadow if `placement` is `'left'` or `'right'`.
+- Fix `n-split` Modified `default-size` panel not updated issue
```
Comment (OrbisK) [inline CHANGELOG.en-US.md:29]:
```suggestion
- Fix `n-split` Modified `default-size` panel not updated issue.
```
😄

**Review by OrbisK — APPROVED** [review]

**Comment by jahnli** [comment]
This part of the API may have to be re-implemented, I will re-open a PR

File: CHANGELOG.en-US.md [hunk 2]
```diff
@@ -53,6 +53,7 @@
 - `n-tree` adds `override-default-node-click-behavior` prop.
 - `n-tree-select` adds `override-default-node-click-behavior` prop.
 - Adds `n-flex` component.
+- `n-split` adds `watch-props` prop, closes [#5526](https://github.com/tusen-ai/naive-ui/issues/5526)
```
Comment (OrbisK) [inline CHANGELOG.en-US.md:56]:
```suggestion
- `n-split` adds `watch-props` prop, closes [#5526](https://github.com/tusen-ai/naive-ui/issues/5526).
```

File: src/split/demos/enUS/index.demo-entry.md [hunk 3]
```diff
@@ -26,6 +26,7 @@ slot.vue
 | min | `number` | `0` | The minimum threshold for splitting, 0-1 is a percentage. | 2.36.0 |
 | max | `number` | `1` | The maximum split threshold, 0-1 is a percentage. | 2.36.0 |
 | resize-trigger-size | `number` | `3` | Size of the resize trigger. | 2.36.0 |
+| watch-props | `Array<defaultSize>` | `undefined` | Default prop names that needed to be watched. Components will be updated after the prop is changed. Note: the `watch-props` itself is not reactive. | 2.36.0 |
```
Comment (OrbisK) [inline src/split/demos/enUS/index.demo-entry.md:29]:
```suggestion
| watch-props | `Array<'defaultSize'>` | `undefined` | Default prop names that needed to be watched. Components will be updated after the prop is changed. Note: the `watch-props` itself is not reactive. | NEXT_VERSION |
```

File: src/split/demos/zhCN/index.demo-entry.md [hunk 4]
```diff
@@ -26,6 +26,7 @@ slot.vue
 | min | `number` | `0` | Split 的分割最小阈值，0-1 代表百分比 | 2.36.0 |
 | max | `number` | `1` | Split 的分割最大阈值，0-1 代表百分比 | 2.36.0 |
 | resize-trigger-size | `number` | `3` | Split 的分隔条大小 | 2.36.0 |
+| watch-props | `Array<defaultSize>` | `undefined` | 需要检测变更的默认属性，检测后组件状态会更新。注意：`watch-props` 本身不是响应式的 | 2.36.0 |
```
Comment (OrbisK) [inline src/split/demos/zhCN/index.demo-entry.md:29]:
```suggestion
| watch-props | `Array<'defaultSize'>` | `undefined` | 需要检测变更的默认属性，检测后组件状态会更新。注意：`watch-props` 本身不是响应式的 | NEXT_VERSION |
```

**Review by OrbisK — CHANGES_REQUESTED** [review]

File: src/split/demos/zhCN/index.demo-entry.md (hunk 4 above)
Comment (jahnli) [reply src/split/demos/zhCN/index.demo-entry.md:29]: :zany_face:

File: CHANGELOG.zh-CN.md [hunk 5]
```diff
@@ -26,6 +26,7 @@
 - 修复 `n-split` 不在卡片中使用没有颜色
 - 修复 `n-card` 的 `footer-class` 属性不生效
 - 修复 `n-tabs` 在 `placement` 为 `'left'` 和 `'right'` 时初始化滚动阴影可能不正确
+- 修复 `n-split` 修改 `default-size` 面板未更新问题
```
Comment (OrbisK) [inline CHANGELOG.zh-CN.md:29]: english translation missing 😄

**Review by OrbisK — APPROVED** [review]

File: src/split/src/Split.tsx [hunk 6]
```diff
@@ -72,6 +74,15 @@ export default defineComponent({
     const isDraggingRef = ref(false)
     const currentSize = ref(props.defaultSize)
 
+    if (props.watchProps?.includes('defaultSize')) {
+      watch(
```
Comment (Sepush) [inline src/split/src/Split.tsx:78]: 这里 watchEffect 应该简单一点

**Review by songjianet — APPROVED** [review]

**Pull request merged**